aspnet/fundamentals/logging.rst

```diff
@@ -1,8 +1,202 @@
+
+.. literalinclude:: logging/sample/src/LoggingSample/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 56-80
+	:emphasize-lines: 4,6-13
+
+The sample above also demonstrates setting the ``MinimumLevel`` on the logger factory. However, this level is simply the default, and can be overridden by individually configured loggers. In this case, the ``sourceSwitch`` is configured to use ``SourceLevels.Critical``, so only ``Critical`` log messages are picked up by the two ``TraceListener``s. When the application is run (using the ``TraceLogging`` environment, on Windows), and a request is made to ``http://localhost:5000/boom``, the following is shown in the console output:
+
+.. image:: logging/_static/console-trace-boom.png
+
+Examining the Application event log in the Windows Event Viewer, the following event has also been logged as a result of this request:
+
+.. image:: logging/_static/eventlog.png
+
+Configuring Other Providers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition to the built-in loggers, you can configure logging to use other providers, such as `NLog <https://github.com/aspnet/Logging/tree/1.0.0-beta6/src/Microsoft.Framework.Logging.NLog>`_ or `Serilog <https://github.com/serilog/serilog-framework-logging>`_. Add the appropriate package to your ``project.json`` file, and then configure it just like any other provider. Typically, these packages should include extension methods on ``ILoggerFactory`` to make it easy to add them. For example, to add support for NLog you would just call:
+
+.. code-block:: c#
+
+	loggerFactory.AddNLog(new NLog.LogFactory());
+
+For Serilog, the code is similar:
+
+.. code-block:: c#
+
+	loggerFactory.AddSerilog();
+
+You can create your own custom providers as well, to support other logging frameworks or your own internal logging requirements. The links above for NLog and Serilog provide good references to help you get started.
+
+Logging Recommendations
+-----------------------
+
+The following are some recommendations you may find helpful when implementing logging in your ASP.NET applications.
+
+1. Log using the correct ``LogLevel``. This will allow you to consume and route logging output appropriately based on the importance of the messages.
+
+2. Log information that will enable errors to be identified quickly. Avoid logging irrelevant or redundant information.
+
+3. Keep log messages concise without sacrificing important information.
+
+4. Although loggers will not log if disabled, consider adding code guards around logging methods to prevent extra method calls and log message setup overhead, especially within loops and performance critical methods.
+
+5. Name your loggers with a distinct prefix so they can easily be filtered or disabled. Remember the ``Create<T>`` extension will create loggers named with the full name of the class.
+
+6. Use Scopes sparingly, and only for actions with a bounded start and end. For example, an MVC action. Avoid nesting many scopes within one another.
```

Comment:
A better example here would be a transaction involving multiple resources. We provide a scope around MVC actions.